content/zh-cn/blog/_posts/2023-01-12-protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,524 @@
+
+```yaml
+---
+# development
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dev-nginx
+  labels:
+    env: dev
+spec:
+  containers:
+  - name: dev-nginx
+    image: nginx
+    resources:
+      requests:
+        memory: "256Mi"
+        cpu: "0.2"
+      limits:
+        memory: ".5Gi"
+        cpu: "0.5"
+```
+
+```yaml
+---
+# preproduction
+apiVersion: v1
+kind: Pod
+metadata:
+  name: preprod-nginx
+  labels:
+    env: preprod
+spec:
+  containers:
+  - name: preprod-nginx
+    image: nginx
+    resources:
+      requests:
+        memory: "1.5Gi"
+        cpu: "1.5"
+      limits:
+        memory: "2Gi"
+        cpu: "2"
+```
+
+```yaml
+---
+# production
+apiVersion: v1
+kind: Pod
+metadata:
+  name: prod-nginx
+  labels:
+    env: prod
+spec:
+  containers:
+  - name: prod-nginx
+    image: nginx
+    resources:
+      requests:
+        memory: "2Gi"
+        cpu: "2"
+      limits:
+        memory: "2Gi"
+        cpu: "2"
+```
+
+<!--
+You can create these pods with the `kubectl create -f <FILE.yaml>` command, and then check their status using the `kubectl get pods` command. You can see if they are up and look ready to serve traffic:
+-->
+你可以使用 `kubectl create -f <FILE.yaml>` 命令创建这些 Pod，然后使用 `kubectl get pods` 命令检查它们的状态。
+你可以查看它们是否已启动并准备好提供流量：
+
+```console
+$ kubectl get pods --show-labels
+NAME            READY   STATUS    RESTARTS   AGE   LABELS
+dev-nginx       1/1     Running   0          55s   env=dev
+preprod-nginx   1/1     Running   0          55s   env=preprod
+prod-nginx      0/1     Pending   0          55s   env=prod
+```
+
+<!--
+Bad news. The pod for the Production environment is still Pending and isn't serving any traffic.
+-->
+坏消息是生产环境的 Pod 仍处于 Pending 状态，并且不能提供任何流量。
+
+<!--
+Let's see why this is happening:
+-->
+让我们看看为什么会发生这种情况：
+
+```console
+$ kubectl get events
+...
+...
+5s          Warning   FailedScheduling   pod/prod-nginx      0/2 nodes are available: 1 Insufficient cpu, 2 Insufficient memory.
+```
+
+<!--
+In this example, there is only one worker node, and that node has a resource crunch.
+-->
+在此示例中，只有一个工作节点，并且该节点存在资源紧缩。
+
+<!--
+Now, let's look at how PriorityClass can help in this situation since prod should be given higher priority than the other environments.
+-->
+现在，让我们看看在这种情况下 PriorityClass 如何提供帮助，因为生产环境应该比其他环境具有更高的优先级。
+
+<!--
+## PriorityClass API
+-->
+## PriorityClass 的 API {#priorityclass-api}
+
+<!--
+Before creating PriorityClasses based on these requirements, let's see what a basic manifest for a PriorityClass looks like and outline some prerequisites:
+-->
+在根据这些需求创建 PriorityClass 之前，让我们看看 PriorityClass 的基本清单是什么样的，并提出一些先决条件：
+
+```yaml
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: PRIORITYCLASS_NAME
+value: 0 # any integer value between -1000000000 to 1000000000 
+description: >-
+  (Optional) description goes here!
+globalDefault: false # or true. Only one PriorityClass can be the global default.
```

Comment:
Consider localizing the comments and the sample description.

Reply:
Translated in the latest update.